### PR TITLE
enable using open mapbox styles

### PR DESCRIPTION
**Pull request by FabianHofmann**
## Changes proposed in this Pull Request
Second try:  Enable using open mapbox styles, [here](https://plot.ly/python/mapbox-layers/) are some examples 

## Checklist

- [x] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- [x] Unit tests for new features were added (if applicable).
- [x] Newly introduced dependencies are added to `environment.yaml`, `environment_docs.yaml` and `setup.py` (if applicable).
- [ ] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.

### CODE REVIEW

File: pypsa/plot.py [hunk 1]
```diff
@@ -605,6 +605,12 @@ def _get_coordinates(network, layouter=None):
         return network.buses["x"], network.buses["y"]
 
 
+_token_required_mb_styles = ['basic', 'streets', 'outdoors', 'light', 'dark',
+                    'satellite', 'satellite-streets']
+
+_open__mb_styles = ['open-street-map', 'white-bg', 'carto-positron',
+                 'carto-darkmatter', 'stamen-terrain', 'stamen-toner', 'stamen-watercolor']
```
Comment (fneum) [inline pypsa/plot.py]: formatting could be tidier.

File: pypsa/plot.py [hunk 2]
```diff
@@ -666,6 +673,18 @@ def iplot(network, fig=None, bus_colors='blue',
         overlapping buses
     mapbox : bool, default False
         Switch to use Mapbox.
+    mapbox_style : str, defaul 'open-street-map'
```
Comment (fneum) [inline pypsa/plot.py]:
```suggestion
    mapbox_style : str, default 'open-street-map'
```

File: pypsa/plot.py [hunk 3]
```diff
@@ -666,6 +673,18 @@ def iplot(network, fig=None, bus_colors='blue',
...
+        Define the mapbox layout style of the interactive plot. If this is set
+        to a mapbox layout, the argument mapbox_token must be a valid Mapbox
```
Comment (fneum) [inline pypsa/plot.py]:
```suggestion
        to a mapbox layout, the argument ``mapbox_token`` must be a valid Mapbox
```

File: pypsa/plot.py (hunk 1 above)
Comment (FabianHofmann) [reply pypsa/plot.py]: I'm trying to figure out what you mean...

Comment (fneum) [reply pypsa/plot.py]: Nothing major. My suggestion is to align the multiline lists:
```python
_token_required_mb_styles = ['basic', 'streets', 'outdoors', 'light', 'dark',
                             'satellite', 'satellite-streets']

_open__mb_styles = ['open-street-map', 'white-bg', 'carto-positron',
                    'carto-darkmatter', 'stamen-terrain', 'stamen-toner', 'stamen-watercolor']
```

Comment (FabianHofmann) [reply pypsa/plot.py]: makes sense

File: pypsa/plot.py [hunk 4]
```diff
@@ -564,9 +564,9 @@ def autogenerate_coordinates(network, assign=False, layouter=None):
         Layouting function from `networkx <https://networkx.github.io/>`_. See
         `list <https://networkx.github.io/documentation/stable/reference/drawing.html#module-networkx.drawing.layout>`_
         of available options. By default coordinates are determined for a
-        `planar layout <https://networkx.github.io/documentation/stable/reference/generated/networkx.drawing.layout.planar_layout.html#networkx.drawing.layout.planar_layout>`_ 
-        if the network graph is planar, otherwise for a 
-        `Kamada-Kawai layout <https://networkx.github.io/documentation/stable/reference/generated/networkx.drawing.layout.kamada_kawai_layout.html#networkx.drawing.layout.kamada_kawai_layout>`_. 
+        `planar layout <https://networkx.github.io/documentation/stable/reference/generated/networkx.drawing.layout.planar_layout.html#networkx.drawing.layout.planar_layout>`_
+        if the network graph is planar, otherwise for a
+        `Kamada-Kawai layout <https://networkx.github.io/documentation/stable/reference/generated/networkx.drawing.layout.kamada_kawai_layout.html#networkx.drawing.layout.kamada_kawai_layout>`_.
```
Comment (FabianHofmann) [inline pypsa/plot.py]: My editor removes trailing spaces, that's why those things here occur

**Pull request merged**